sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -162,6 +162,9 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
         toSQL(p.right),
         p.condition.map(" ON " + _.sql).getOrElse(""))
 
+    case h @ Hint(_, _, s @ SubqueryAlias(alias, p @ Project(_, _: SQLTable))) =>
```

Comment:
The new way looks cleaner, but this looks like a special handling. Can you write a code comment to explain when it is being used?